UnityProjects/MRTKDevTemplate/Assets/Scripts/DialogSpawnerExample.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.UX;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    public class DialogSpawnerExample : MonoBehaviour
+    {
+        public DialogSpawner dialogSpawner;
+
+        public void Awake()
+        {
+            if (dialogSpawner == null)
+            {
+                dialogSpawner = GetComponent<DialogSpawner>();
+            }
+        }
+
+        public void SpawnDialogFromCode()
+        {
+            IDialog dialog = dialogSpawner.Get()
```

Comment:
Any reason we don't allow for setting some of these fields in the dialogSpawner's arugments?

`i.e. dialogSpawner.Spawn(header, body, positiveName, negativeName)`

Or something similar. I'm a bit worried that it's not intuitive that you have to set Header, body etc, text after spawning the dialog. 

At the very least, the text in those fields feel like they can be provided to spawn the dialog initially.

Reply:
That would require extra boilerplate and management code in the dialogSpawner; when you call the `.SetHeader`, etc, you're actually talking directly to the dialog. In addition, this allows us to not have a bunch of overrides/different overridden method signatures on the spawner; the spawner isn't in charge of controlling/binding data or actions, it's just the lifecycle management of the view.

It's designed to work like this: https://developer.android.com/guide/fragments/dialogs

Reply:
In addition, subclassed Dialog types can add new controls/parameters. If we provided aliases to those in the spawner, subclassed Dialogs wouldn't have the equivalent option in the spawner (because the spawner wouldn't be subclassed)